client/src/pages/board/components/Column.jsx

```diff
@@ -53,10 +64,33 @@ const Tasks = memo(({tasks}) =>
 export default function Column({column, index, tasks, setUpdate}) {
   const classes = useStyles();
   const [open, setOpen] = useState(false);
-
+  const [title, setTitle] = useState(column.title);
+  const [isEditing, setEditing] = useState(false);
+  const {dispatch} = useContext(BoardContext);
   const handleClickOpen = () => {
     setOpen(true);
   };
+  const editTitle = (e) => {
+    e.preventDefault();
+    setTitle(e.target.value);
+  };
+
+  const finishEdit = () => {
+    setEditing(false);
+    updateColumn(column.id, {name: title});
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      finishEdit();
+    }
+  };
+
+  const deleteCol = () => {
+    boardActions.deleteColumn(column.id, dispatch);
+    removeColumn(column.id);
+  };
+
```

Comment:
This looks good! Thanks for adding these crud functionalities.

Reply:
Looking at this, I do see why you'd call the api here on the FE component since it's so simple to change the title vs the order of cards. I think having column api called here is fine vs the dnd actions.